cmd/gitquery/server.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/src-d/gitquery"
+	"github.com/src-d/gitquery/internal/function"
+	sqle "gopkg.in/src-d/go-mysql-server.v0"
+	"gopkg.in/src-d/go-mysql-server.v0/server"
+	"gopkg.in/src-d/go-vitess.v0/mysql"
+)
+
+// CmdServer defines server command
+type CmdServer struct {
+	cmd
+
+	Git      string `short:"g" long:"git" description:"Path where the git repositories are located, one per dir"`
+	Host     string `short:"h" long:"host" default:"localhost" description:"Host where the server is going to listen"`
+	Port     int    `short:"p" long:"port" default:"3306" description:"Port where the server is going to listen"`
+	User     string `short:"u" long:"user" default:"root" description:"User name used for connection"`
+	Password string `short:"P" long:"password" default:"" description:"Password used for connection"`
+
+	engine *sqle.Engine
+	name   string
+}
+
+func (c *CmdServer) buildDatabase() error {
+	if c.engine == nil {
+		c.engine = sqle.New()
+	}
+
+	c.print("opening %q repository...\n", c.Git)
+
+	var err error
+
+	pool := gitquery.NewRepositoryPool()
+	err = pool.AddDir(c.Git)
+	if err != nil {
+		return err
+	}
+
+	c.engine.AddDatabase(gitquery.NewDatabase(c.name, &pool))
+	function.Register(c.engine.Catalog)
+	return nil
+}
+
+// Execute starts the server
+func (c *CmdServer) Execute(args []string) error {
+	if err := c.buildDatabase(); err != nil {
+		return err
+	}
+
+	auth := mysql.NewAuthServerStatic()
+	auth.Entries[c.User] = []*mysql.AuthServerStaticEntry{
+		{Password: c.Password},
+	}
+
+	hostString := fmt.Sprintf("%v:%v", c.Host, c.Port)
```

Comment:
Do you think is valuable use `net.JoinHostPort` instead of `fmt.Sprintf`?

Reply:
Thanks for the tip. JoinHostPort supports IPv6 addresses so it's better.